src/main/java/com/zufar/icedlatte/security/api/UserAuthenticationService.java

```diff
@@ -54,14 +58,36 @@ public UserAuthenticationResponse authenticate(final UserAuthenticationRequest r
             log.warn("Invalid credentials for user's account with email = '{}'", userEmail, exception);
             loginFailureHandler.handle(userEmail);
             throw new BadCredentialsException(String.format("Invalid credentials for user's account with email = '%s'", userEmail), exception);
-
         } catch (LockedException exception) {
             log.warn("User's account with email = '{}' is locked", userEmail, exception);
             throw new UserAccountLockedException(userEmail, userAccountLockoutDurationMinutes);
+        } catch (Exception exception) {
+            log.error("Error occurred during authentication", exception);
+            throw exception;
+        }
+    }
+
+    public UserAuthenticationResponse authenticate(final UserDetails userDetails, String userEmail) {
+        try {
+            String jwtRefreshToken = jwtTokenProvider.generateRefreshToken(userDetails);
+            String jwtToken = jwtTokenProvider.generateToken(userDetails);
+            log.info("Generated JWT token for user with email = '{}'", userEmail);
 
+            resetLoginAttemptsService.reset(userEmail);
+
+            return new UserAuthenticationResponse(jwtToken, jwtRefreshToken);
+        }
```

Comment:
please make a line break for the bracket

Reply:
Fixed